post-publish-processor/src/main/scala/org/sunbird/job/postpublish/helpers/BatchCreation.scala

```diff
@@ -139,17 +140,25 @@ trait BatchCreation {
     }
   }
 
-  def addCertTemplateToBatch(contextId: String, batchId: String, contextType: String)(implicit cassandraUtil: CassandraUtil, config: PostPublishProcessorConfig, httpUtil: HttpUtil) = {
+  def addCertTemplateToBatch(eData: java.util.Map[String, AnyRef], batchId: String, contextType: String)(implicit cassandraUtil: CassandraUtil, config: PostPublishProcessorConfig, httpUtil: HttpUtil) = {
+    val contextId = eData.get(config.identifier).asInstanceOf[String]
     logger.info("Adding cert template to batch:" + batchId + ", contextId: " + contextId + ", contextType: " + contextType)
     val selectQuery = QueryBuilder.select().all().from(config.sunbirdKeyspaceName, config.sbSystemSettingsTableName)
     var certTemplateId = config.defaultCertTemplateId
     var certTemplateAddPath = config.batchAddCertTemplateAPIPath
     var reqIdKey = "courseId"
 
     if ("Event".equalsIgnoreCase(contextType)) {
-      certTemplateId = config.defaultEventCertTemplateId
-      certTemplateAddPath = config.batchAddCertTemplateAPIPathForEvent
-      reqIdKey = "eventId"
```

Comment:
I would suggest to keep all three lines instead of removing... 

remove the else part which is added at line #157